Upgrade EUI to v102.3.0

mgadewoll opened the pull request:
`102.2.0` ⏩ `102.3.0`

[Questions? Please see our Kibana upgrade FAQ.](https://github.com/elastic/eui/blob/main/wiki/eui-team-processes/upgrading-kibana.md#faq-for-kibana-teams)

## Changes

>[!IMPORTANT]
This release contains the visual refresh updates for buttons ⏹️ 💅 
These have already been available for QA ahead of time on [this PR](https://github.com/elastic/kibana/pull/220093).

1. **EUI changes**: Adds all Visual Refresh related EUI changes on button components:

- `EuiButton`
- `EuiButtonEmpty`
- `EuiButtonIcon`
- `EuiButtonGroup`
- `EuiFilterGroup`/`EuiFilterButton`

2. **EUI button component API updates:** Updates instances of `EuiFilterButton` usages that were missing `isSelected` prop or that should use the new `isToggle` prop to ensure the expected visual and semantic output as toggle button.

<details><summary><b>Examples of new toggle filter buttons</b></summary>

| before | after |
|---|---|
| ![Screenshot 2025-05-05 at 16 18 13](https://github.com/user-attachments/assets/65dcb4f1-27c2-4e73-8d0e-2702de8c35e3) | ![Screenshot 2025-05-02 at 13 48 28](https://github.com/user-attachments/assets/54ec723b-ff3f-49b7-ad6c-0d804fcd8a82) |
| ![Screenshot 2025-05-05 at 16 19 30](https://github.com/user-attachments/assets/65bcef9a-c06b-46e4-91e1-698a76b1dbc2) | ![Screenshot 2025-05-02 at 14 02 46](https://github.com/user-attachments/assets/bf05d9f9-0eba-4733-87f6-d7abdaa53bfa) |
| ![Screenshot 2025-05-05 at 16 21 50](https://github.com/user-attachments/assets/8f4a6592-9dc0-4b69-8ba5-aca0a3c26f69) | ![Screenshot 2025-05-02 at 14 08 07](https://github.com/user-attachments/assets/34d78faa-4137-4650-a4a9-f71cbde64054) |
</details>


3. **Custom styling updates:** Updates custom styling overrides on EUI button components in selected **high-visibility places** (discover, dashboard, lens) to ensure expected Visual Refresh output. (This is a first iteration step to ensure expected design. In the future we want to replace custom buttons with proper EUI components)

<details><summary><b>Updated custom button styling</b></summary>

| Solution/Area | After |
|---|---|
| Discover(classic) | ![Screenshot 2025-05-30 at 14 45 01](https://github.com/user-attachments/assets/93a46e2a-4989-4bcb-99f2-5eafa4483629) |
| Discover(ES/QL) | ![Screenshot 2025-05-30 at 14 45 11](https://github.com/user-attachments/assets/8fc390d4-fdb0-4377-851c-b3835dc9a14e) |
| Dashboard(Create Visualization) | ![Screenshot 2025-05-30 at 14 47 21](https://github.com/user-attachments/assets/a5895309-9b48-494d-b4b4-e91ca41e316b) |
| Maps: Aligned map toolbar button hover  | ![Screenshot 2025-04-30 at 17 19 11](https://github.com/user-attachments/assets/2d467f6b-ab50-46b1-b393-ed1e6c5d5548) |

</details>

## Package updates

### `@elastic/eui`

## [`v102.3.0`](https://github.com/elastic/eui/releases/v102.3.0)

- Added support for `focusTrapProps.returnFocus` on `EuiFlyout` ([#8716](https://github.com/elastic/eui/pull/8716))
- Updated `EuiDataGridToolbarControl` hover styles by removing text-decoration and changing the badge background-color to ensure enough contrast ([#8670](https://github.com/elastic/eui/pull/8670))
- Added prop `isToggle` on `EuiFilterButton` to switch between regular and toggle button ([#8652](https://github.com/elastic/eui/pull/8652))
- Updated `hasActiveFilters` on `EuiFilterButton` to only control highlighting filters, not a visual selected state ([#8652](https://github.com/elastic/eui/pull/8652))
- Updated `EuiFilterButton` to ensure `isSelected` prop applies both semantic and visual states ([#8652](https://github.com/elastic/eui/pull/8652))
- Updated visual styling of `EuiButtonGroup`, `EuiFilterGroup` and `EuiFilterButton` ([#8652](https://github.com/elastic/eui/pull/8652))
- Updated `EuiNotificationBadge` border radius to `2px` ([#8652](https://github.com/elastic/eui/pull/8652))
- Updated `EuiBreadcrumbs` text color for `type="application"` ([#8652](https://github.com/elastic/eui/pull/8652))
- Added new `refresh` design variant for button components: ([#8595](https://github.com/elastic/eui/pull/8595))
  - `EuiButton`
  - `EuiButtonEmpty`
  - `EuiButtonIcon`
- Added `flags.buttonVariant` with value `classic` to `euiThemeAmsterdam` ([#8595](https://github.com/elastic/eui/pull/8595))
- Added new button background component tokens to Amsterdam theme: ([#8595](https://github.com/elastic/eui/pull/8595))
  - `background{color}Hover`
  - `background{color}Active`
  - `backgroundFilled{color}Hover`
  - `backgroundFilled{color}Active`
  - `backgroundEmpty{color}Hover`
  - `backgroundEmpty{color}Active`

**Bug fixes**

- Fixed an issue with `EuiDataGrid` where navigating cells with simultaneous key presses would result in a crash in React 18 legacy mode ([#8698](https://github.com/elastic/eui/pull/8698))

**Accessibility**

- Improved the accessibility experience of `EuiMarkdownEditorFooter` by fixing the ARIA attributes of the `syntax help` modal. ([#8702](https://github.com/elastic/eui/pull/8702))

### `@elastic/eui-theme-borealis`

## [`v1.1.0`](https://github.com/elastic/eui/releases/v1.1.0)

- Added token `filterButtonBadgeBackgroundHover` ([#8652](https://github.com/elastic/eui/pull/8652))
- Updated values for tokens `buttonGroupFocusColor` and `buttonGroupBackgroundDisabledSelected` ([#8652](https://github.com/elastic/eui/pull/8652))
- Added `flags.buttonVariant` with value `refresh` to `euiThemeBorealis` ([#8595](https://github.com/elastic/eui/pull/8595))
- Added new button background component tokens: ([#8595](https://github.com/elastic/eui/pull/8595))
  - `background{color}Hover`
  - `background{color}Active`
  - `backgroundFilled{color}Hover`
  - `backgroundFilled{color}Active`
  - `backgroundEmpty{color}Hover`
  - `backgroundEmpty{color}Active`
- Updated `backgroundLightText` token value to `shade120` ([#8595](https://github.com/elastic/eui/pull/8595))

<!--ONMERGE {"backportTargets":["8.19","9.0"]} ONMERGE-->

mgadewoll committed:
test: update tests and snapshots

mgadewoll committed:
test: update selector

jeramysoucy committed:
Updates Kibana security custom style components

mgadewoll committed:
chore: update eui packages

mgadewoll committed:
refactor: update filterButton usages

- adds isToggle and Selected for toggle buttons; adds missing isSelected on expander buttons

mgadewoll committed:
refactor: update custom button styles to align with Visual Refresh button updates

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/303978)
* Commit: ffd6e4f67c48807808ad7f391e8061f9180706e1

### Failed CI Steps
* [Explore - Security Solution Cypress Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/303978#01972fc1-ea3c-4f54-9025-3ffce0a82d3f)

cc @mgadewoll

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303978","number":303978,"commit":"ffd6e4f67c48807808ad7f391e8061f9180706e1"}],"number":303978}
buildkite-pr-comment-->

elasticmachine commented:
Pinging @elastic/eui-team (EUI)

rmyz reviewed (approved):
`x-pack/solutions/observability/plugins/observability_shared/public/components/field_value_suggestions/field_value_selection.tsx` changes LGTM

elasticmachine commented:
Pinging @elastic/fleet (Team:Fleet)

sabarasaba reviewed (approved):
code owners changes lgtm

ThomThomson reviewed (approved):
Changes LGTM! Ran this branch locally and looked through all the changes to button design and made sure everything worked the same. Left a few comments, but nothing urgent.

mgadewoll commented on src/platform/plugins/shared/presentation_util/public/components/field_picker/field_type_filter.tsx:
```diff
@@ -49,7 +49,7 @@ export function FieldTypeFilter({
       data-test-subj="toggleFieldFilterButton"
       iconType="arrowDown"
-      isSelected={fieldTypesValue.length > 0}
+      isSelected={isPopoverOpen}
```
@ThomThomson The internal API for **EuiFilterButton** changed slightly. Previously `hasActiveFilters` would control the filter badge highlight _and_ the selected visual state which was not quite right. With the introduction of a real toggle variant we changed this to be aligned. `isSelected` now controls the selected state visually and semantically in Borealis (Amsterdam should still behave as it did previously).

I do see what you mean though about the bold text. That's actually a small unexpected change on EUI side. There is a change from font weight `400` to `450` when `hasActiveFilters` is toggled 🙈 Thanks for the catch! 🎉 

https://github.com/user-attachments/assets/cd33db4b-e23f-4d5f-8ba5-051f35d2bbf9

https://github.com/user-attachments/assets/dd679871-fd69-48b0-b022-68a968050044

I'll update that on EUI side, and it'll be fixed with the next release.

peteharverson reviewed (approved):
ML changes LGTM. Ran locally and all looked good. Like the changes to `EuiFilterButton`!

justinkambic reviewed (commented):
There's one file with a change I don't understand. Please explain or delete.

mgadewoll committed:
chore: cleanup random leftover

mgadewoll commented on x-pack/solutions/observability/plugins/synthetics/public/apps/synthetics/components/common/components/filter_status_button.tsx:
```diff
@@ -44,7 +46,7 @@ export const FilterStatusButton = ({
       color={color}
     >
-      {content}
+      {content}fooooo
```
Removed in https://github.com/elastic/kibana/pull/222149/commits/97a1efb40d0bc703b3d719ceeffe82529d843209. Thanks for the catch 🙏

justinkambic reviewed (approved):
Management changes LGTM, code review only.

kc13greiner reviewed (approved):
Security Changes LGTM!

criamico reviewed (approved):
Fleet changes LGTM

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/305055)
* Commit: 431685b94cbb60a07941fec41fdf1ca318d23e2b
* [Cloud Deployment](https://kibana-pr-222149.kb.us-west2.gcp.elastic-cloud.com)

### History
* :broken_heart: [Build #304779](https://buildkite.com/elastic/kibana-pull-request/builds/304779) failed 97a1efb40d0bc703b3d719ceeffe82529d843209
* :yellow_heart: [Build #303978](https://buildkite.com/elastic/kibana-pull-request/builds/303978) was flaky ffd6e4f67c48807808ad7f391e8061f9180706e1

cc @mgadewoll

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305055","number":305055,"commit":"431685b94cbb60a07941fec41fdf1ca318d23e2b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304779","number":304779,"commit":"97a1efb40d0bc703b3d719ceeffe82529d843209"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303978","number":303978,"commit":"ffd6e4f67c48807808ad7f391e8061f9180706e1"}],"number":305055}
buildkite-pr-comment-->

thomheymann reviewed (commented):
Thanks for all the work on this, everything looking good from our side except for one small issue:

The button on the Dataset Quality page doesn't look right. The icon doesn't get inverted correctly and the button has a hover effect which looks odd in this context - It should be rendered more like a link with an icon.

<img width="384" alt="Screenshot 2025-06-04 at 10 33 32" src="https://github.com/user-attachments/assets/407778a0-6783-48e7-909d-f61a8d82df87" />

Page: `/app/management/data/data_quality`

georgianaonoleata1904 reviewed (approved):
ResponseOps changes LGTM

thomheymann commented:
> > Thanks for checking. Ignoring the issue with icon in dark mode, how do we get the styling of the button correct so that it's rendered like in link with an underline instead of the hover effect? This is rendered correctly on [edge](https://edge-lite-oblt.kb.us-west2.gcp.elastic-cloud.com/app/management/data/data_quality/?pageState=(filters:(fullNames:!f,inactive:!t,integrations:!(),namespaces:!(),qualities:!(),timeRange:(from:now-24h,refresh:(pause:!t,value:60000),to:now),types:!(logs)),table:(page:0,rowsPerPage:10,sort:(direction:asc,field:title)),v:1))
> 
> @thomheymann Ah right, sorry I skipped over this part 🙈 So the change is expected from EUI side for the underlying `EuiButtonEmpty` component in use. Looking at this I rather think that the usage of `EuiHeaderLink` [here](https://github.com/elastic/kibana/blob/main/x-pack/platform/plugins/shared/dataset_quality/public/components/dataset_quality/table/dataset_quality_details_link.tsx#L44) is actually contextually not correct (as we're not in a header). Instead to update this and return to a link styling we could use a `EuiButtonEmpty` directly (`EuiHeaderLink` is just a wrapper) with `flush="both"`.

Brilliant, sounds good!

mgadewoll committed:
refactor: ensure DatasetQualityDetailsLink uses link styling

ryankeairns reviewed (approved):
Awesome work @mgadewoll .
Thank you for the proactive adjustments to all the filter buttons :)

rylnd reviewed (approved):
Detection Engine changes (to a single panel currently behind a FF) LGTM.

albertoblaz reviewed (approved):
Cloud Security Posture changes OK

agusruidiazgd reviewed (approved):
Thanks for this awesome work! LGTM for explore files

eokoneyo reviewed (approved):
Shared UX changes LGTM

nikitaindik reviewed (approved):
Thanks! Reviewed and tested Rule Mgmt owned pages. Everything looks and works great. New buttons are awesome! 👍

mgadewoll committed:
chore: cleanup

- removes visual border between label and control for optionList, this is to align with the previous state but will change in the next iteration for inputs

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/306301)
* Commit: f323156154d2d6deb29a15f6f4cd2a58d40e7cbe
* [Cloud Deployment](https://kibana-pr-222149.kb.us-west2.gcp.elastic-cloud.com)

### History
* :yellow_heart: [Build #305255](https://buildkite.com/elastic/kibana-pull-request/builds/305255) was flaky a3ba38c032c11c73f2e818d8a690305cade2d87d
* :yellow_heart: [Build #305055](https://buildkite.com/elastic/kibana-pull-request/builds/305055) was flaky 431685b94cbb60a07941fec41fdf1ca318d23e2b
* :broken_heart: [Build #304779](https://buildkite.com/elastic/kibana-pull-request/builds/304779) failed 97a1efb40d0bc703b3d719ceeffe82529d843209
* :yellow_heart: [Build #303978](https://buildkite.com/elastic/kibana-pull-request/builds/303978) was flaky ffd6e4f67c48807808ad7f391e8061f9180706e1

cc @mgadewoll

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"running","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/306301","number":306301,"commit":"f323156154d2d6deb29a15f6f4cd2a58d40e7cbe"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305255","number":305255,"commit":"a3ba38c032c11c73f2e818d8a690305cade2d87d"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/305055","number":305055,"commit":"431685b94cbb60a07941fec41fdf1ca318d23e2b"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/304779","number":304779,"commit":"97a1efb40d0bc703b3d719ceeffe82529d843209"},{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/303978","number":303978,"commit":"ffd6e4f67c48807808ad7f391e8061f9180706e1"}],"number":306301}
buildkite-pr-comment-->

mgadewoll commented:
@elasticmachine merge upstream

elasticmachine committed:
Merge branch 'main' into eui/v102.3.0

mgadewoll commented:
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|9.0|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/223091">](https://github.com/elastic/kibana/pull/223091)|
|✅|8.19|[<img src="https://img.shields.io/github/pulls/detail/state/elastic/kibana/223098">](https://github.com/elastic/kibana/pull/223098)|

Note: Successful backport PRs will be merged automatically after passing CI.

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--- Backport version: 10.0.0 -->
